woocommerce/woocommerce
# Add/ Checkout step logging

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #51387 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure logging is enabled in `/wp-admin/admin.php?page=wc-status&tab=logs&view=settings`, and clear any existing logs before proceeding.
2. Place two orders: one using checkout blocks, one using shortcode.
3. Verify on `/wp-admin/admin.php?page=wc-status&tab=logs` that there are no `place-order-debug-xxxxxxx` registered (logs get cleared on successful/handled failed orders)
4. Introduce some problems in the shortcode order flow by using the following snippet:
```php
add_action( 'woocommerce_checkout_order_processed', 'custom_die_on_order_processed', 10, 3 );
function custom_die_on_order_processed( $order_id, $posted_data, $order ) {
	die( 'Processing stopped for testing purposes.' );
}
```
5. Verify a `place-order-debug-xxxxxxx` log file is presented, with a list of completed steps carrying context data.
6. Repeat steps 4 and 5 for a Blocks placed order using the following snippet:
woocommerce_store_api_checkout_update_order_from_request
```php
add_action( 'woocommerce_store_api_checkout_update_order_from_request', 'failing_too', 10, 3 );
function failing_too( $orde, $request ) {
	die( 'Processing stopped for testing purposes.' );
}
```


### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Added step logging on the shortcode and Store API place order flow.
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
